tests/test_signals.py

```diff
@@ -0,0 +1,379 @@
+    assert signal.sampling_rate == 44100
+    assert signal.fft_norm == 'none'
+
+
+def test_impulse_with_user_parameters():
+    """Test impulse with custom delay, amplitude and sampling rate"""
+    signal = pfs.impulse(3, 1, 2, 48000)
+    npt.assert_allclose(signal.time, np.atleast_2d([0, 2, 0]))
+
+
+def test_impulse_multi_channel():
+    """Test multi channel impulse."""
+    # test with array and number
+    signal = pfs.impulse(3, [0, 1], 1)
+    ref = np.atleast_2d([[1, 0, 0], [0, 1, 0]])
+    npt.assert_allclose(signal.time, ref)
+    # test with two arrays
+    signal = pfs.impulse(3, [0, 1], [1, 2])
+    ref = np.atleast_2d([[1, 0, 0], [0, 2, 0]])
+    npt.assert_allclose(signal.time, ref)
+
+
+def test_impulse_float():
+    """Test sine signal with float duration."""
+    signal = pfs.impulse(441.8)
+    assert signal.n_samples == 441
+
+
+def test_white_noise_with_defaults():
+    """Test wite noise with default parameters."""
+    signal = pfs.white_noise(100)
+
+    assert isinstance(signal, Signal)
+    assert signal.sampling_rate == 44100
+    assert signal.fft_norm == "rms"
+    npt.assert_allclose(np.sqrt(np.mean(signal.time**2, axis=-1)), 1)
+
+
+def test_white_noise_with_user_parameters():
+    """Test wite noise with amplitude and sampling rate."""
+    signal = pfs.white_noise(100, 2, 48000)
+
+    assert signal.sampling_rate == 48000
+    npt.assert_allclose(np.sqrt(np.mean(signal.time**2, axis=-1)), 2)
+
+
+def test_white_noise_multi_channel():
+    """Test wite noise with amplitude and sampling rate."""
+    rms = [[1, 2, 3], [4, 5, 6]]
+    signal = pfs.white_noise(100, rms)
+    npt.assert_allclose(np.sqrt(np.mean(signal.time**2, axis=-1)), rms)
+
+
+def test_white_noise_seed():
+    """Test passing seeds to the random generator."""
+    a = pfs.white_noise(100, seed=1)
+    b = pfs.white_noise(100, seed=1)
+    assert a == b
+
+    a = pfs.white_noise(100)
+    b = pfs.white_noise(100)
+    with pytest.raises(AssertionError):
+        assert a == b
+
+
+def test_white_noise_float():
+    """Test sine signal with float duration."""
+    signal = pfs.white_noise(441.8)
+    assert signal.n_samples == 441
+
+
+def test_pink_noise_with_defaults():
+    """
+    Test only the defaults because pink noise uses the same private functions
+    as white noise.
+    """
+    signal = pfs.pink_noise(100)
+
+    assert isinstance(signal, Signal)
+    assert signal.sampling_rate == 44100
+    assert signal.fft_norm == "rms"
+    npt.assert_allclose(np.sqrt(np.mean(signal.time**2, axis=-1)), 1)
+
+
+def test_pink_noise_rms_spectrum():
+    """
+    Test for constant energy across filters of constant relative band width.
+    """
+    # filtered pink noise
+    # (use only center octaves, because the spectrum is less stochastic there)
+    signal = pfs.pink_noise(5000, seed=1)
+    signal = pff.fractional_octave_bands(signal, 1, freq_range=(1e3, 16e3))
+    # check if stdandard deviation is less then 1%
```

Comment:
Where does this value come from?

Reply:
Oh sorry never replied - this is hand tuned. The longer the noise the smaller this tolerance could be :)

Reply:
Show me that nice and tasty Frickelfaktor :)